src/api/hooks/useGetANS.ts

```diff
@@ -26,43 +27,53 @@ function getFetchNameUrl(
 
 export function useGetNameFromAddress(address: string, shouldCache = false) {
   const [state, _] = useGlobalState();
-  const [name, setName] = useState<string | undefined>();
 
-  useEffect(() => {
-    const cachedName = getLocalStorageWithExpiry(address);
-    if (cachedName) {
-      setName(cachedName);
-      return;
-    }
+  const cachedName = getLocalStorageWithExpiry(address);
+  if (cachedName) {
+    return cachedName;
+  }
+
+  const queryResult = useQuery<string, ResponseError>({
+    queryKey: ["ANSName", address, shouldCache, state.network_name],
+    queryFn: () => genANSName(address, shouldCache, state.network_name),
+  });
 
-    const primaryNameUrl = getFetchNameUrl(state.network_name, address, true);
-    if (primaryNameUrl !== undefined) {
-      const fetchData = async () => {
-        const {name: primaryName} = await fetchJsonResponse(primaryNameUrl);
-
-        if (primaryName) {
-          if (shouldCache) {
-            setLocalStorageWithExpiry(address, primaryName, TTL);
-          }
-          setName(primaryName);
-        } else {
-          const nameUrl =
-            getFetchNameUrl(state.network_name, address, false) ?? "";
-          const {name} = await fetchJsonResponse(nameUrl);
-          if (shouldCache) {
-            setLocalStorageWithExpiry(address, name, TTL);
-          }
-          setName(name);
-        }
-      };
-
-      fetchData().catch((error) => {
-        console.error("ERROR!", error, typeof error);
-      });
+  // return empty string if ans name not found to prevent useQuery complaining about undefined return
+  return queryResult.data === "" ? undefined : queryResult.data;
+}
+
+async function genANSName(
+  address: string,
+  shouldCache: boolean,
+  networkName: NetworkName,
+): Promise<string> {
+  const primaryNameUrl = getFetchNameUrl(networkName, address, true);
+
+  if (!primaryNameUrl) {
+    return "";
+  }
+
+  try {
+    const {name: primaryName} = await fetchJsonResponse(primaryNameUrl);
+
+    if (primaryName) {
+      if (shouldCache) {
+        setLocalStorageWithExpiry(address, primaryName, TTL);
+      }
+      return primaryName;
+    } else {
+      const nameUrl = getFetchNameUrl(networkName, address, false) ?? "";
```

Comment:
this seems kinda bad for error noise. We're defaulting to an empty string and then make an http request to empty string which will certainly lead to an error. Let's just return undefined and skip the request part as a result.